### DISCUSSION TITLE
v0.0.6

### DISCUSSION THREAD

**Discussion by LefterisXris** [LefterisXris/CodeTour#70] [Announcements]
### Added
- Enhance File path handling for more flexible Navigation (includes directory path as prefix)
- Tour's summary on Step's description including the current and the total number of Steps of the Tour
- Fix potential conflict with the existence of files with the same name by prompting a picker dialog to the user
- Disable Prev/Next buttons when no Prev/Next Step exists accordingly
- Introduce validation logic to make sure available tours are valid (currently checks for target file existence)
- Added a default Virtual Onboarding Assistant as a sample Tour. Enabled by default

### Highlight

The great feature of this release is the Virtual Onboarding Assistant, which is a default Tour (can be enabled/disabled from settings) that provides a step-by-step guide that if followed, can result a very good understanding of any project!
Give it a try and let me know your thoughts!

<hr /><em>This discussion was created from the release <a href='https://github.com/LefterisXris/CodeTour/releases/tag/v0.0.6'>v0.0.6</a>.</em>

**Comment by vsethi13** [comment]
How do I set the directory property on a step's definition? I see no option to do so in the step editor.
I am using CodeTour v0.0.6 on IntelliJ Idea 2022.2.4

**Reply by LefterisXris** [reply]
Hello @vsethi13! Indeed the directory property is not editable through Step Editor. 
However, it is supported (for compatibility reasons with VS Code's plugin as requested) and can be configured directly through Tour's source code. 

To do so, follow these steps:
- Open the `Tours Navigation` panel
- `Right click` on the Tour you want to edit
- Select `Jump to .tour Source` option
- Find the Step for which you want to configure a directory
- Add a directory field, in the following form: `"directory": "[YOUR/DIRECTORY/PATH]"`
  - Example: `"directory": "src/main/resources/META_INF"`

Full Example:

```json
{
  "id": "70aaa0f5-3fb7-4309-a9bf-97afeb28aa9b",
  "tourFile": "sampleTour.tour",
  "title": "Sample Tour",
  "description": "A Sample Tour",
  "steps": [
    {
      "title": "A Cool Step 1",
      "description": "A Step on a file whose name exists twice on the project",
      "directory": "src/main/resources",
      "file": "logback.xml",
      "line": 2
    },
    {
      "title": "A Cool Step 2",
      "description": "A Step on a file whose name exists twice on the project",
      "directory": "src/main/resources/logging",
      "file": "logback.xml",
      "line": 2
    }
  ]
}
```

Note: the **directory** property makes the path for the Step's target file more specific. This is going to be enhanced, so that a directory can be pointed by a Step (without any file), meaning that when a user clicks on that Step, the directory will be selected on the `Project Explorer` Tool Window Pane. This will soon be implemented. You can track it through #41

**Comment by oculus42** [comment]
Hello! I just discovered this plugin but ran into a version incompatibility when trying to install on WebStorm:

<img width="486" alt="CodeTour-install-error" src="https://user-images.githubusercontent.com/1731657/207709957-c8702acf-6089-4244-bd05-3fead70e1ca9.png">

WebStorm 2022.3
Build #WS-223.7571.168, built on November 28, 2022

I'm definitely interested in testing it out once it's updated for the new version!

—Cheers!

**Reply by LefterisXris** [reply]
Hello @oculus42! Thanks for pointing it out, apparently I was not aware of it! 
I will prepare a new version compatible with all 2023.* Jetbrains products and will let you know!
Thanks again!

**Reply by LefterisXris** [reply]
@oculus42  Compatibility with latest Jetbrain products is now implemented.
You may upgrade to CodeTour version v0.0.7.
